openforcefield/typing/engines/smirnoff/forcefield.py

```diff
@@ -817,11 +817,33 @@ def _load_smirnoff_data(self, smirnoff_data, allow_cosmetic_attributes=False):
             Whether to permit non-spec kwargs in smirnoff_data.
         """
         import packaging.version
+
+        # Check that the SMIRNOFF version of this data structure is supported by this ForceField implementation
+
+        if "SMIRNOFF" in smirnoff_data.keys():
+            version = smirnoff_data['SMIRNOFF']['version']
+        elif "SMIRFF" in smirnoff_data.keys():
```

Comment:
You can remove the `keys()` here to make use of the constant time lookup.

Reply:
If I get some free time I'm  going to make an Andrea-bot to remind me of this from now on.